### PR TITLE
update rust 1.64.0

### PR DESCRIPTION
**Pull request by uvd**
### Description
update rust 1.64.0
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4570)

### CODE REVIEW

File: crates/aptos-rosetta/src/account.rs [hunk 1]
```diff
@@ -209,10 +209,7 @@ async fn get_balances(
         if let Some(currencies) = maybe_filter_currencies {
             let mut currencies: HashSet<Currency> = currencies.into_iter().collect();
             // Remove extra currencies not requested
-            balances = balances
-                .into_iter()
-                .filter(|balance| currencies.contains(&balance.currency))
-                .collect();
+            balances.retain(|balance| currencies.contains(&balance.currency));
```
Comment (gregnazario) [inline crates/aptos-rosetta/src/account.rs:212]: Oh that's a cool new one

File: mempool/src/tests/common.rs [hunk 2]
```diff
@@ -142,9 +142,7 @@ pub(crate) fn batch_add_signed_txn(
     transactions: Vec<SignedTransaction>,
 ) -> Result<()> {
     for txn in transactions.into_iter() {
-        if let Err(e) = add_signed_txn(pool, txn) {
-            return Err(e);
-        }
+        add_signed_txn(pool, txn)?;
```
Comment (gregnazario) [inline mempool/src/tests/common.rs:145]: :/ this kinda reads much worse imo

Very easy to miss the ?

File: storage/backup/backup-cli/src/backup_types/epoch_ending/backup.rs [hunk 3]
```diff
@@ -57,8 +57,7 @@ impl EpochEndingBackupController {
     pub async fn run(self) -> Result<FileHandle> {
         info!(
             "Epoch ending backup started, starting from epoch {}, until epoch {} (excluded).",
-            start_epoch = self.start_epoch,
-            end_epoch = self.end_epoch,
+            self.start_epoch, self.end_epoch,
```
Comment (gregnazario) [inline storage/backup/backup-cli/src/backup_types/epoch_ending/backup.rs:60]: So, this looks like an issue with the structured logging setup.

This should probably be before the message
```
start_epoch = self.start_epoch,
end_epoch = self.end_epoch,
```

and this should be after
```
self.start_epoch, self.end_epoch
```
@msmouse is this your intention?

**Review by gregnazario — APPROVED** [review]
lgtm, some weird differences but mostly looked like some good cleanup

File: storage/backup/backup-cli/src/backup_types/epoch_ending/backup.rs (hunk 3 above)
Comment (davidiw) [reply storage/backup/backup-cli/src/backup_types/epoch_ending/backup.rs:60]: this shouldn't be changed, this adjusts how the (un)structured logging works.

Comment (uvd) [reply storage/backup/backup-cli/src/backup_types/epoch_ending/backup.rs]: > this shouldn't be changed, this adjusts how the (un)structured logging works.

Well, I changed it back

File: mempool/src/tests/common.rs (hunk 2 above)
Comment (banool) [reply mempool/src/tests/common.rs:145]: This is much more idiomatic imo, most everyone who codes Rust knows to look out for `?`. Perhaps a comment to clear it up.

Comment (uvd) [reply mempool/src/tests/common.rs:145]: @banool  @gregnazario   This is the only way to pass the test without changing the rules of lint

**Pull request closed without merging**